source/extensions/filters/http/jwt_authn/verifier.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "extensions/filters/http/jwt_authn/authenticator.h"
+#include "extensions/filters/http/jwt_authn/verify_context.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+
+class Verifier;
+typedef std::unique_ptr<Verifier> VerifierPtr;
+
+/**
+ * Supports verification of JWTs with configured requirments.
+ */
+class Verifier {
+public:
+  virtual ~Verifier() {}
+
+  // Verify all tokens on headers, and signal the caller with callback.
+  virtual void verify(VerifyContext& context) PURE;
+  // Set the verifier's parent group verifier.
+  virtual void registerParent(Verifier* parent) PURE;
```

Comment:
this registerParent() should not be in the interface.  it should be implementation detail.

Reply:
removed.